lib/ansible/modules/cloud/google/gcp_dns_resource_record_set.py

```diff
@@ -0,0 +1,466 @@
+      service_account_file: "{{ gcp_cred_file }}"
+      scopes:
+        - https://www.googleapis.com/auth/ndev.clouddns.readwrite
+      state: present
+  register: managed_zone
+- name: create a resource record set
+  gcp_dns_resource_record_set:
+      name: 'www.testzone-4.com.'
+      managed_zone: "{{ managed_zone }}"
+      type: 'A'
+      ttl: 600
+      target:
+        - 10.1.2.3
+        - 40.5.6.7
+      project: testProject
+      auth_kind: service_account
+      service_account_file: /tmp/auth.pem
+      scopes:
+        - https://www.googleapis.com/auth/ndev.clouddns.readwrite
+      state: present
+'''
+
+RETURN = '''
+    name:
+        description:
+            - For example, www.example.com.
+        returned: success
+        type: str
+    type:
+        description:
+            - One of valid DNS resource types.
+        returned: success
+        type: str
+    ttl:
+        description:
+            - Number of seconds that this ResourceRecordSet can be cached by
+              resolvers.
+        returned: success
+        type: int
+    target:
+        description:
+            - As defined in RFC 1035 (section 5) and RFC 1034 (section 3.6.1)
+        returned: success
+        type: list
+    managed_zone:
+        description:
+            - A reference to ManagedZone resource
+        returned: success
+        type: dict
+'''
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible.module_utils.gcp_utils import navigate_hash, GcpSession, GcpModule, GcpRequestException
+import json
+import copy
+import datetime
+import time
+
+################################################################################
+# Main
+################################################################################
+
+
+def main():
+    """Main function"""
+
+    module = GcpModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            name=dict(required=True, type='str'),
+            type=dict(required=True, type='str'),
+            ttl=dict(type='int'),
+            target=dict(type='list'),
+            managed_zone=dict(required=True, type='dict'),
+        )
+    )
+
+    # Replace the dictionary of results from other GCP modules with the
+    # proper value only.
+    module.params['managed_zone'] = module.params['managed_zone']['name']
+
+    state = module.params['state']
+    kind = 'dns#resourceRecordSet'
+
+    fetch = fetch_wrapped_resource(module, 'dns#resourceRecordSet',
+                                   'dns#resourceRecordSetsListResponse',
+                                   'rrsets')
+    changed = False
+
+    if fetch:
+        if state == 'present':
+            if is_different(module, fetch):
+                fetch = update(module, self_link(module), kind, fetch)
```

Comment:
Does changed need to be set to True here?

Reply:
Yes, good find! Fixed.

Reply:
(it's the next line)